src/server/routes/configuration/wallets/update.ts

```diff
@@ -68,76 +63,66 @@ export async function updateWalletsConfiguration(fastify: FastifyInstance) {
       },
     },
     handler: async (req, res) => {
-      switch (req.body.type) {
-        case WalletType.local:
-          await updateConfiguration({
-            awsAccessKeyId: null,
-            awsSecretAccessKey: null,
-            awsRegion: null,
-            gcpApplicationProjectId: null,
-            gcpKmsLocationId: null,
-            gcpKmsKeyRingId: null,
-            gcpApplicationCredentialEmail: null,
-            gcpApplicationCredentialPrivateKey: null,
-          });
-          break;
-        case WalletType.awsKms:
-          if (
-            !req.body.awsAccessKeyId ||
-            !req.body.awsSecretAccessKey ||
-            !req.body.awsRegion
-          ) {
-            throw createCustomError(
-              "Please specify all AWS KMS configuration.",
-              StatusCodes.BAD_REQUEST,
-              "BAD_REQUEST",
-            );
-          }
+      if ("awsAccessKeyId" in req.body) {
+        if (
+          !req.body.awsAccessKeyId ||
+          !req.body.awsSecretAccessKey ||
+          !req.body.awsRegion
+        ) {
+          throw createCustomError(
+            "Please specify all AWS KMS configuration.",
+            StatusCodes.BAD_REQUEST,
+            "BAD_REQUEST",
+          );
+        }
+        await updateConfiguration({
+          awsAccessKeyId: req.body.awsAccessKeyId,
+          awsSecretAccessKey: req.body.awsSecretAccessKey,
+          awsRegion: req.body.awsRegion,
+        });
+      }
 
-          await updateConfiguration({
-            awsAccessKeyId: req.body.awsAccessKeyId,
-            awsSecretAccessKey: req.body.awsSecretAccessKey,
-            awsRegion: req.body.awsRegion,
-            gcpApplicationProjectId: null,
```

Comment:
Ok good, this was the main part that would delete other wallet creds when adding new configs.